.github/workflows/main.yml

```diff
@@ -0,0 +1,112 @@
+          mkdir -p ./data/certs
+          openssl genpkey -algorithm RSA -pkeyopt rsa_keygen_bits:2048 -out ./data/certs/local-ca.key
+          openssl req -x509 -new -nodes -key ./data/certs/local-ca.key -sha256 -days 3650 -out ./data/certs/local-ca.pem -subj "/C=US/ST=Denial/L=Springfield/O=Dis/CN=www.example.com"
+          openssl req -x509 -new -nodes -key ./data/certs/local-ca.key -sha256 -days 3650 -out ./data/certs/factory_ca.pem -subj "/C=US/ST=Denial/L=Springfield/O=Dis/CN=www.example.com"
+          openssl genpkey -algorithm RSA -pkeyopt rsa_keygen_bits:2048 -out ./data/certs/server.key
+          openssl req -new -key ./data/certs/server.key -out ./data/certs/server.csr -subj "/C=US/ST=Denial/L=Springfield/O=Dis/CN=www.example.com"
+          echo subjectAltName=DNS:www.example.com,DNS:example.org > ./data/certs/san.cnf
+          openssl x509 -req -in ./data/certs/server.csr -CA ./data/certs/local-ca.pem -CAkey ./data/certs/local-ca.key -CAcreateserial -out ./data/certs/tls-crt -days 365 -sha256 -extfile ./data/certs/san.cnf
+          rm -rf ./data/certs/server.key ./data/certs/server.csr ./data/certs/san.cnf
+          chmod +r ./data/certs/*   
+
+      - name: Run Docker Compose Up
+        run: |
+          docker-compose up --force-recreate --detach
+
+      - name: Verify call of the /sign endpoint
+        run: |
+          max_attempts=12
+          attempt=1
+          while [[ $attempt -le $max_attempts ]]
+          do
+            response=$(curl -X POST http://localhost:80/sign -o /dev/null -w '%{http_code}' || true)
+            echo "Attempt $attempt: Response code: $response"
+            if [[ "$response" -eq 400 ]]; then
+              echo "Expected 400 Bad Request error received."
+              exit 0
+            else
+              echo "Unexpected response code received: $response"
+              if [[ $attempt -eq $max_attempts ]]; then
+                echo "Maximum attempts reached without receiving expected response."
+                exit 1
+              fi
+              echo "Waiting for 3 seconds before next attempt..."
+              sleep 3
+            fi
+            ((attempt++))
+          done
+
+
+      - name: Call fake-lmp-device-register with invalid values
+        run: |
+          mkdir -p ./var/sota/
+          set +e
+            output=$(python3 fake-lmp-device-register --registration-url "http://localhost:80/sign" --factory fake-test-sample 2>&1)
+            exit_code=$?
+          set -e
+          if [ $exit_code -ne 0 ]; then
+            echo "Non-zero exit code: $exit_code"
+            if echo "$output" | grep -q '500 Internal Server Error'; then
+              echo "Expected 500 Internal Server Error when doing the request to https://api.foundries.io/ota/devices/ to register the device."
+            else
+              echo "Unexpected failure occurred."
+              exit 1
+            fi
+          fi
+          echo "Output:"
+          echo "$output"
```

Comment:
Instead of these various bash based tests for error scenarios, can we have just one test for the happy path?

I mean, call the `fake-lmp-device-register` with good enough parameters to succeed.

It is fine if that is not hard to achieve; but then I'd rather have no CI test.
It seems strange to have the CI which expects the program to fail.

Reply:
> I mean, call the fake-lmp-device-register with good enough parameters to succeed.

It should fail when we reach out:

```
 r = requests.put(
            "https://api.foundries.io/ota/devices/", headers=headers, json=data
        )
``` 

I do not think that we should add valid certs here.
The tests made in this way allow us to ensure that the container build, runs, and the fake client works and, indeed test the app.py up to do the request.

Reply:
> We have only one check where we expected an error.
> When we call the fake we expected to get 500 when we call 

This is exactly the problem.
This CI tests that things fail; but it does not check if things succeed.
That is a strange thing to have; checking for 500 error does not look sane at best.

> It would only succeed by using real and valid certs.

You already created the CAs for the registration server; which are expected to be valid.

> Also, we would need to have a factory to use as mock.

Not really; registration server can work in fully detached mode, where it knows nothing about the factory.

> So, I do not think that we should do that.

I agree we need to have a CI in ideal world.
But this CI needs to verify the happy path; not that things fail with application error.

Reply:
@doanac @camilamacedo86 I'd recommend to put this on hold.

@camilamacedo86 did a great job to make sure that `fake-lmp-device-register` works.
Having a nearly zero activity in this repo that should be fine for the next few years.

Thus, I consider tha amount of effort invested into this CI too big and the thing it checks too small.